design/one-pager-default-resource-class.md

```diff
@@ -88,66 +116,237 @@ spec:
-kind: ResourceClass
+apiVersion: database.crossplane.io/v1alpha1
+kind: MySQLInstance
 metadata:
-  name: cloud-nosql
-  namespace: crossplane-system
-parameters:
-  ...
-defaultForClaimKinds:
-- nosqlinstance.database.crossplane.io
-- dynamodbinstance.storage.crossplane.io
-provisioner: dynamodbinstance.database.aws.crossplane.io/v1alpha1
-providerRef:
-  name: aws-provider
-reclaimPolicy: Delete
+  name: mysql-claim
+spec:
+  classRef:
+    kind: RDSInstanceClass
+    apiVersion: database.aws.crossplane.io/v1alpha1
+    name: standard-mysql
+    namespace: crossplane-system
+  writeConnectionSecretToRef:
+    name: rdsmysql
+  engineVersion: "5.6"
 ```
 
-And a resource claim that defaulted to the resource class above could look like this in the general case:
+To continue to provide the same level of portability for claims that was
+originally present, two enhancements to the `Policy` model can be made:
+
+1. Allow for multiple policies per namespace.
+1. Denote the default policy via label.
+
+### Multiple Policies Per Namespace
+
+Currently, default class controllers, which act on claims that omit a
+`classRef`, will fail to reconcile if multiple `Policies` exist within the
+claim's namespace. For example, if two `MySQLInstancePolicies` are created in
+namespace `crossplane-system`, then a `MySQLInstance` claim created in
+`crossplane-system` that omits a `classRef` will not be assigned either one of
+the classes referenced by the two `MySQLInstancePolicy` objects respectively.
+
+In order to reintroduce the ability to select a class of service in claim, the
+following steps can be taken:
+
+1. Update all `Policy` types to include a `classRef` field instead of a
+   `defaultClassRef` field. This would require updating the `crossplane-runtime`
+   embeddable struct
+   [`Policy`](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/apis/core/v1alpha1/resource.go#L93),
+   then bumping the dependency in core Crossplane.
+1. Remove the `classRef` field from all claims entirely. This would require the
+   same steps as above but with
+   [`ResourceClaimSpec`](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/apis/core/v1alpha1/resource.go#L49).
+1. Add a `providerRef` field to all claims using the same method detailed above.
+   This would require updating the same steps as above also with
+   [`ResourceClaimSpec`](https://github.com/crossplaneio/crossplane-runtime/blob/e4d61ee2805af680baf16fc2a1d8f79538d0f9bb/apis/core/v1alpha1/resource.go#L49).
+1. Alter all claim controllers (which live in provider stacks) to only reconcile
+   claims that have a `policyRef` to a `Policy` that has a `classRef` of that
+   kind's strongly-typed class. For instance, the `RDSInstance` claim controller
+   will only reconcile `MySQLInstance` claims that have `policyRef` to a
+   `MySQLInstancePolicy` that has a `classRef` to an `RDSInstanceClass`. This
```

Comment:
Do we imagine the mapping of `RDSInstance` to `RDSInstanceClass` as a generated/dynamic mapping?

Reply:
https://godoc.org/github.com/crossplaneio/crossplane-runtime/pkg/resource#NewClaimReconciler

This mapping is currently specified when instantiating each resource claim controller's generic reconciler. I imagine under this proposal we'd also have to specify which which policy kind the reconciler should use. We could do so using the same pattern, but might want to rethink it since the `NewClaimReconciler` signature is getting fairly long.